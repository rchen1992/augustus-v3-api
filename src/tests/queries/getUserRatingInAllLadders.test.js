const mockUserRepo = require('@tests/mockRepos/user');
const { testQuery } = require('@tests/utils');

testQuery({
    id: "Get a user's ratings and rating deltas in all of their ladders",
    query: `
        {
            user(id: 1) {
                id
                ladders {
                    id
                    ladderName
                    userRating
                    userRatingDelta
                }
            }
        }
    `,
    variables: {},
    context: () => {
        return {
            repos: {
                user: mockUserRepo,
            },
        };
    },
    expected: undefined,
});
