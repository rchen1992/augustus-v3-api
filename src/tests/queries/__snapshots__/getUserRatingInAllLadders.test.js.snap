// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Get a user's ratings and rating deltas in all of their ladders 1`] = `
Object {
  "data": Object {
    "user": Object {
      "id": "1",
      "ladders": Array [
        Object {
          "id": "1",
          "ladderName": "Tagpad",
          "userRating": 160,
          "userRatingDelta": 30,
        },
        Object {
          "id": "2",
          "ladderName": "Kamba",
          "userRating": 100,
          "userRatingDelta": 30,
        },
        Object {
          "id": "3",
          "ladderName": "Skyvu",
          "userRating": 140,
          "userRatingDelta": 20,
        },
        Object {
          "id": "4",
          "ladderName": "Thoughtworks",
          "userRating": 200,
          "userRatingDelta": -5,
        },
        Object {
          "id": "5",
          "ladderName": "Yakijo",
          "userRating": 100,
          "userRatingDelta": -10,
        },
      ],
    },
  },
}
`;
